jscomp/others/js_array.ml

```diff
@@ -17,61 +17,122 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
-
 type 'a t = 'a array
+type 'a array_like
+type 'a array_iter = 'a array_like (* don't think this is very useful to implement wihtout language support *)
+
+external make : int -> unit Js.undefined array = "Array" [@@bs.new]
```

Comment:
I included it for the ability to do things like `make 10 |> mapi (fun _ i -> i)`, but I agree `Array.make` is better.